### DISCUSSION TITLE
Why sometimes you use type mixed and sometimes you don't?

### DISCUSSION THREAD

**Discussion by DominikStyp** [worksome/request-factories#27] [General]
What is the point of using type mixed?

Sometimes it's in the closure's attribute: 
https://github.com/worksome/request-factories/blob/main/src/Actions/CreateFactoryResultSteps/ResolveNestedRequestFactories.php#L21

Sometimes it's not:
https://github.com/worksome/request-factories/blob/main/src/Support/Result.php#L49

Isn't also better to use the expected array keys instead of mixed like here:
https://github.com/worksome/request-factories/blob/main/src/Support/ConfigBasedFinder.php#L13

Where you only use two keys from the config, so it could easily be:
```
/**
 * @param array{namespace:string|null, path:string|null} $config
 */
 public function __construct(private array $config)
 {
 }
```
...and this provides some useful information, which mixed doesn't

**Comment by lukeraymonddowning** [comment]
Depends on my mood. I find the mixed type sometimes helps me remember I shouldn't assume what to expect. 

If you want to PR something here, feel free :-)
